### ISSUE TITLE
Multithreaded file and string consumption

### ISSUE THREAD

**Issue by Adamtaranto** [dib-lab/oxli#22]
How fast can we make kmer counting in Rust? 

@ctb, are there examples of this from Sourmash or other libraries?

**Comment by ctb** [comment]
yes, it's actually really easy ;). The trick is to make sure we're using `iter` and closures wherever possible; then we will just change `iter` to `par_iter`, include rayon's prelude, and voila.

This was next on my list after https://github.com/dib-lab/oxli/pull/10 gets finished off.

**Comment by Adamtaranto** [comment]
Might also save some time if we revcomp the full DNA seq once and pass a sliding window backwards through it (as in sourmash seqtohashes) instead of calculating rc for every kmer.

**Comment by Adamtaranto** [comment]
@ctb what do you think about making the user specify a thread number for rayon? I think it will try to use all available by default.

**Comment by Adamtaranto** [comment]
Some other Rust kmer counting projects for ideas:
- [Krust](https://github.com/suchapalaver/krust/tree/main)
- [KmerUtils](https://github.com/jean-pierreboth/kmerutils) uses bloom filters
- [Kmers](https://docs.rs/kmers/latest/kmers/)
